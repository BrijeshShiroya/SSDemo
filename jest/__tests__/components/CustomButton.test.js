import 'react-native';
import React from 'react';
import CustomButton from '../../../app/components/CustomButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('CustomButton renders correctly', () => {
  const customButtonComponent = renderer.create(<CustomButton />).toJSON();
  expect(customButtonComponent).toMatchSnapshot();
});

it('CustomButton render with loading', () => {
  const customButtonComponent = renderer
    .create(<CustomButton isLoading />)
    .toJSON();
  expect(customButtonComponent).toMatchSnapshot();
});

it('CustomButton disable', () => {
  const customButtonComponent = renderer.create(<CustomButton disabled />);
  expect(customButtonComponent).toMatchSnapshot();
});

it('CustomButton with title', () => {
  const customButtonComponent = renderer.create(
    <CustomButton title={'Header'} />
  );
  expect(customButtonComponent).toMatchSnapshot();
});
