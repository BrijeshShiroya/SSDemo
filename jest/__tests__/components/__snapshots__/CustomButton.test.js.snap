// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton disable 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "height": 78.85714285714286,
      "justifyContent": "center",
      "opacity": 1,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirNext-Bold",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  />
</View>
`;

exports[`CustomButton render with loading 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(248, 156, 33)",
      "borderRadius": 4,
      "height": 78.85714285714286,
      "justifyContent": "center",
      "opacity": 1,
      "width": 300,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`CustomButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(248, 156, 33)",
      "borderRadius": 4,
      "height": 78.85714285714286,
      "justifyContent": "center",
      "opacity": 1,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirNext-Bold",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  />
</View>
`;

exports[`CustomButton with title 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(248, 156, 33)",
      "borderRadius": 4,
      "height": 78.85714285714286,
      "justifyContent": "center",
      "opacity": 1,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "AvenirNext-Bold",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    Header
  </Text>
</View>
`;
